scale-tests/scale_test.py

```diff
@@ -22,35 +22,43 @@
 
 
 def submit_job(dispatcher):
+    print("T dispatcher: {}".format(dispatcher))
     dispatcher_name, dispatcher_role, driver_role = dispatcher
 
-    args = ["--conf", "spark.cores.max=1",
+    args = ["--conf", "spark.cores.max=4",
+            "--conf", "spark.executor.cores=1",
             "--conf", "spark.mesos.containerizer=mesos",
-            "--conf", "spark.mesos.role={}".format(driver_role),
-            "--conf", "spark.mesos.executor.docker.image=mesosphere/spark-dev:931ca56273af913d103718376e2fbc04be7cbde0",
+            "--conf", "spark.mesos.executor.docker.image=mesosphere/spark-dev:f5dd540adffd9ab9e3e826e48d22e39ebc296567-aee120236ee37309892ddbee42acf0028096947f",
             # use Hector's image
-            "--conf", "spark.port.maxRetries=32"  # setting to allow up to 32 drivers on same node
+            "--conf", "spark.port.maxRetries=32",  # setting to allow up to 32 drivers on same node
             #"--conf", "spark.mesos.driverEnv.SPARK_USER=root", # Run as root on centos
+            "--supervise",
+            #"--conf", "spark.mesos.rejectOfferDuration=1000s",
+            "--conf", "spark.mesos.driver.failoverTimeout=30",
+            "--conf", "spark.mesos.task.labels=jpmc_test_id:spark_batch_run001"
```

Comment:
Should we make the task label configurable? Is this code not also used by the streaming jobs?

Reply:
Yes, that makes sense. No, there's currently a separate script for streaming jobs. @samvantran looked into merging it but found it difficult. The big difference is that the streaming jobs get kicked off once and are long-running. The batch jobs get submitted continuously and end an hour after they're launched.

Reply:
I've copied over mostly every conf used in this script to the other one (didn't see this label one until now